thavalon-server/src/game/role.rs

```diff
@@ -0,0 +1,131 @@
+use std::fmt::{self, Write};
+
+use rand::prelude::*;
+
+use super::{Card, PlayerId, Players};
+
+/// A THavalon role
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum Role {
+    Merlin,
+    Lancelot,
+    Percival,
+    Tristan,
+    Iseult,
+
+    Mordred,
+    Morgana,
+    Maelegant,
+    Agravaine,
+}
+
+impl Role {
+    pub fn is_good(self) -> bool {
+        use Role::*;
+        match self {
+            Merlin | Lancelot | Percival | Tristan | Iseult => true,
+            Mordred | Morgana | Maelegant | Agravaine => false,
+        }
+    }
+
+    pub fn is_evil(self) -> bool {
+        !self.is_good()
+    }
+
+    pub fn is_lover(self) -> bool {
+        match self {
+            Role::Tristan | Role::Iseult => true,
+            _ => false,
+        }
+    }
+
+    pub fn is_assasinatable(self) -> bool {
+        match self {
+            Role::Merlin | Role::Tristan | Role::Iseult => true,
+            _ => false,
+        }
+    }
+
+    pub fn can_play(self, card: Card) -> bool {
+        // The life of an Agravaine is a simple one
+        if self == Role::Agravaine {
+            card == Card::Fail
+        } else {
+            match card {
+                Card::Success => true,
+                Card::Reverse => matches!(self, Role::Lancelot | Role::Maelegant),
+                Card::Fail => self.is_evil(),
+            }
+        }
+    }
+
+    /// Create role information for a player, `me`, given all `players` in the game.
+    pub fn generate_info<R: Rng>(self, _rng: &mut R, me: PlayerId, players: &Players) -> String {
+        // TODO: let this be markdown or something?
```

Comment:
We can discuss - I guess what side should handle formatting, front end or back end? I'd argue that front end should handle formatting, and get back separate things for info, such as:

- team
- description (who you are/summary of what you know)
- info (what you know)
- powers (what you can do)

Reply:
Formatting on the front-end sounds good to me

Reply:
Took a stab at modeling info, let me know what you think